addons/slack/template/index.js

```diff
@@ -12,59 +20,56 @@ module.exports = async (ctx) => {
   if (ctx.url.endsWith('events/ingest')) {
     Sdk.debug('Received Events API event');
 
-    const match = ctx.headers.authorization.match(/^bearer\s+(.+)$/i);
-    const token = match && match[1];
-    if (!token || token !== ctx.configuration.fusebit_events_token) {
-      Sdk.debug('Missing or incorrect auth token');
-      return { status: 403 };
-    }
+    if (ctx.caller.permissions) {
+      try {
+        if (ctx.method === 'POST') {
+          let event = ctx.body;
+          Sdk.debug('Valid event', event);
 
-    try {
-      if (ctx.method === 'POST') {
-        let event = ctx.body;
-        Sdk.debug('Valid event', event);
+          // Attach all event handlers off a nested `slack` property
+          let handlers = slack.slack;
+          if (handlers) {
+            const functions = Object.getOwnPropertyNames(handlers)
+              .filter((p) => typeof handlers[p] === 'function')
+              .filter((f) => f == event.type || f == 'all');
+            const client = await getStorageClient(ctx);
+            const promises = functions.map((f) =>
+              handlers[f](sdk, event, {
+                ...ctx.configuration,
+                storage: client,
+              })
+            );
 
-        // Attach all event handlers off a nested `slack` property
-        let handlers = slack.slack;
-        if (handlers) {
-          const functions = Object.getOwnPropertyNames(handlers)
-            .filter((p) => typeof handlers[p] === 'function')
-            .filter((f) => f == event.type || f == 'all');
-          const client = Sdk.getStorageClient(ctx);
-          const promises = functions.map((f) =>
-            handlers[f](sdk, event, {
-              ...ctx.configuration,
-              storage: client,
-            })
-          );
+            Sdk.debug('Invoking handlers', functions);
+            await Promise.all(promises);
 
-          Sdk.debug('Invoking handlers', functions);
-          await Promise.all(promises);
-
-          Sdk.debug('All handlers completed');
+            Sdk.debug('All handlers completed');
+          }
+          return {
+            status: 201,
+          };
+        } else {
+          return {
+            status: 400,
+            body: { message: 'Unsupported HTTP request type' },
+          };
         }
-        return {
-          status: 201,
-        };
-      } else {
-        return {
-          status: 400,
-          body: { message: 'Unsupported HTTP request type' },
-        };
+      } catch (e) {
+        Sdk.debug('Dispatching to event handlers failed', e);
+        return { status: 500 };
```

Comment:
Is there a more useful error message that could be included here, to speed debugging of failing handlers?

Reply:
Added